pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,162 @@
+package buildchain
+
+import (
+	"fmt"
+	"os"
+
+	// "code.google.com/p/gographviz"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	_ "github.com/openshift/origin/pkg/image/api"
+	val "github.com/openshift/origin/pkg/image/api/validation"
+)
+
+// node is a representation of a node inside a tree
+type node struct {
+	name     string
+	tags     []string
+	children []*node
+}
+
+// String helps in pretty-printing a tree
+func (root *node) String() string {
+	tree := ""
+	tree += root.name
+
+	for _, n := range root.children {
+		tree += "(" + n.String() + ")"
+	}
+
+	return tree
+}
+
+const longDescription = `Experimental command
+
+Output build dependencies of a specific image repository.
+
+Examples:
+
+    $ openshift ex build-chain [image-repository]
+`
+
+func NewCmdBuildChain(f *clientcmd.Factory, parentName, name string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s [image-repository]", name),
+		Short: "Output build dependencies of a specific image repository",
+		Long:  longDescription,
+		Run: func(cmd *cobra.Command, args []string) {
+			imageRepoName := cmdutil.GetFlagString(cmd, "imagerepo")
+			if imageRepoName == "" {
+				usageError(cmd, "Must pass an image repository name with the --imagerepo (-i) flag")
```

Comment:
Shouldn't you just take the first argument as the image repo name? (line 45 implies that you do)